internal/internal_task_handlers.go

```diff
@@ -1687,6 +1689,22 @@ func isCommandMatchEvent(d *commandpb.Command, e *historypb.HistoryEvent, obes [
 			return false
 		}
 		return true
+
+	case enumspb.COMMAND_TYPE_SCHEDULE_NEXUS_OPERATION:
+		if e.GetEventType() != enumspb.EVENT_TYPE_NEXUS_OPERATION_SCHEDULED {
+			return false
+		}
+		eventAttributes := e.GetNexusOperationScheduledEventAttributes()
+		commandAttributes := d.GetScheduleNexusOperationCommandAttributes()
+
+		if eventAttributes.GetService() != commandAttributes.GetService() || eventAttributes.GetOperation() != commandAttributes.GetOperation() {
```

Comment:
To confirm, we allow endpoint to be changed without failing determinism but not service or operation? One might expect service and endpoint to be the same kind of thing, or at least it may be confusing to explain to users that half of the Nexus client creation can be changed and the other half cannot. I think it should be both endpoint and service or neither. An argument for both is that endpoints can be changed server side.

Reply:
Unresolving to continue conversation. I think you should either allow the two things that create a client to be changed or disallow the two things that create a client from being changed. Having one of those two changeable is confusing. If you're looking for a task queue analogy, I think that's endpoint + service and if we had an "activity client", creating it would probably have the task queue like the nexus client has endpoint + service.